src/Pos/Txp/Toil/Types.hs

```diff
@@ -61,13 +61,13 @@ utxoF = later formatUtxo
 
 data BalancesView = BalancesView
     { _bvStakes :: !(HashMap StakeholderId Coin)
-    , _bvTotal  :: !Coin
+    , _bvTotal  :: !(Maybe Coin)
```

Comment:
Because it doesn't make any sense to have `Coin` here. This data type is a _modifier_.

>  where will we use this Maybe?

For example, in `Default` instance, as you can see already.

>  what is goal of Maybe here, why not just 0 coins?

0 coins means that total stake is 0 coins.
`Nothing` means that total stake is same as in underlying storage.
Using 0 instead of `Nothing` is not elegant.